core/src/main/java/org/elasticsearch/index/query/SimpleQueryStringBuilder.java

```diff
@@ -205,7 +205,9 @@ protected void doWriteTo(StreamOutput out) throws IOException {
         out.writeOptionalString(minimumShouldMatch);
         if (out.getVersion().onOrAfter(Version.V_5_1_1)) {
             out.writeOptionalString(settings.quoteFieldSuffix());
-            out.writeOptionalBoolean(useAllFields);
+            if (out.getVersion().before(Version.V_6_0_0_beta2)) {
+                out.writeOptionalBoolean(null);
```

Comment:
I agree. I pushed https://github.com/elastic/elasticsearch/pull/26145/commits/456ceac1eac240e119291bd7277f356a1b8be6a5 to fix this.